src/endpoints/http/collections/export/index.js

```diff
@@ -0,0 +1,143 @@
+ */
+function getCollectionStreams(collections, db, format, raw) {
+  return collections.map(name => {
+    const collection = db.collection(name);
+    const cursor = collection.find({}, { raw: raw });
+    cursor.filename = `${name}.${format}`;
+    return cursor.stream();
+  });
+}
+
+/**
+ * Parses all collections requested for export in the requested format.
+ *
+ * @param {Stream} streams - An array of streams containing each collection's documents.
+ * @param {String} format - The requested format to export the collections.
+ * @returns {Stream} - An array of streams containing each collection's documents parsed in the requested format.
+ */
+function setParsers(streams, format) {
+  return streams.map(collection => {
+    try {
+      const parsedCollections = parsers[format](collection);
+      return parsedCollections;
+    } catch (err) {
+      throw new UnsupportedMediaError();
+    }
+  });
+}
+
+/**
+ * Pipe's the zip file out to a writeable stream to be downloaded.
+ *
+ * @param {Stream} zipFile - A zip file stream containing each collection's documents parsed in the requested format.
+ * @param {Stream} out - Response stream containing the zip file.
+ * @returns {Promise}
+ */
+function exportZip(zipFile, out) {
+  return new Promise((resolve, reject) => {
+    zipFile
+      .pipe(out)
+      .on('finish', resolve)
+      .on('error', reject);
+  });
+}
+
+/**
+ * Exports collection(s) for a given app.
+ *
+ * @param {db} db - The db connection.
+ * @param {String} reqCollections - The String array of collections requested to export.
+ * @param {String} format - The requested format to export the collections. Supports bson, json and csv.
+ * @param {Stream} out - Response stream containing the zip file.
+ * @returns Promise.
+ */
+function exportHandler(db, collectionNames, format, out) {
+  return getCollectionsSize(db, collectionNames)
+    .then(sizes => {
+      if (getTotalSize(sizes) >= GB) {
+        throw new Error("Cannot export collections larger than a gigabyte");
+      }
+      const streams = getCollectionStreams(collectionNames, db, format, format === 'bson');
+      const parsedCollections = setParsers(streams, format);
+      return Promise.all(parsedCollections);
+    })
+    .then(collections => archive(collections))
+    .then(zipFile => exportZip(zipFile, out));
+}
+
+/**
+ * Invokes exportHandler with requested collections or else all collections.
+ *
+ * @param {db} db - The db connection.
+ * @param {String} reqCollections - The String array of collections requested to export.
+ * @param {String} format - The requested format to export the collections. Supports bson, json and csv.
+ * @param {Stream} out - The output stream of the zip entries.
+ * @returns exportHandler - function to handle the exporting of collections.
+ */
+export default function exportCollections(db, reqCollections, format, out) {
+  if (!reqCollections.length) {
+    return db
```

Comment:
This can probably be extracted into a function

```javascript
return getAllCollectionNames(db)
  .then(collectionNames => {
    return exportHandler(db, collectionNames, format, out);
  })
```
```javascript
function getAllCollectionNames(db) {
  return db
     .listCollections()
     .map(collection => collection.name.split('.').pop())
     .toArray()
     .then(names => {
         return names.filter(name => name !== 'indexes' & name !== 'users');
      });
}
```